pkg/ingester/ingester_v2.go

```diff
@@ -452,3 +461,63 @@ func (i *Ingester) closeAllTSDB() {
 	i.userStatesMtx.Unlock()
 	wg.Wait()
 }
+
+// openExistingTSDB walks the user tsdb dir, and opens a tsdb for each user. This may start a WAL replay, so we limit the number of
+// concurrently opening TSDB.
+func (i *Ingester) openExistingTSDB(ctx context.Context) error {
+	wg := &sync.WaitGroup{}
+	openGate := gate.New(i.cfg.TSDBConfig.MaxOpeningTSDBOnStartup)
+
+	err := filepath.Walk(i.cfg.TSDBConfig.Dir, func(path string, info os.FileInfo, err error) error {
+		if path == i.cfg.TSDBConfig.Dir { // Nothing to do for root
+			return nil
+		}
+
+		// Top level directories are assumed to be user TSDBs
+		if info.IsDir() {
+
+			userID := info.Name()
+			f, err := os.Open(path)
+			if err != nil {
+				level.Error(util.Logger).Log("msg", "unable to open user TSDB dir", "err", err, "user", userID)
```

Comment:
What do you think if we also log the `path`, for a quicker understanding?

Reply:
Added.